tests/basics.test.ts

```diff
@@ -157,8 +158,9 @@ test('choices should refine the type of the field to a union', () => {
 
 test('misconfigured spec', () => {
   // Validation throws with different error if spec is invalid
-  // @ts-expect-error This misuse should be a type error
-  expect(() => cleanEnv({ FOO: 'asdf' }, { FOO: {} }, makeSilent)).toThrow()
+  expect(() => {
+    expectTypeOf(cleanEnv({ FOO: 'asdf' }, { FOO: {} }, makeSilent)).toBeNever()
+  }).toThrow()
```

Comment:
Personally I think I prefer the `// @ts-expect-error` comments even though they're a bit of a hack. They read as more explicit to me

Reply:
This was _not_ an opinion change. It was required due to the way type inference work now; it will output a `never` type in place of a type error. Which is, functionally, equivalent since you wouldn't be able to do anything useful with the returned object.

### No type error

`// @ts-expect-error` will not work here:

![image](https://user-images.githubusercontent.com/3646758/203769139-c203e310-9dfc-4bc7-a287-2a1ba43bdf8e.png)

### Type error at property access site
![image](https://user-images.githubusercontent.com/3646758/203769413-9385a975-32ba-4eff-b0d8-5b232cdd1fe0.png)

### Discussion

However, we could if you prefer write something like this:

```ts
expect(() => {
  // @ts-expect-error - This misuse should be a type error
  cleanEnv({ FOO: 'asdf' }, { FOO: {} }, makeSilent).FOO
}).toThrow()
```

But it's worth to mention that `never` is probably the best fit, because this is exactly what it was designed for:

> The never type represents values which are never observed. In a return type, this means that the function throws an exception or terminates execution of the program.

Source: [the official TypeScript doc](https://www.typescriptlang.org/docs/handbook/2/functions.html#never).